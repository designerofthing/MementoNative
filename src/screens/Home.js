import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  StyleSheet,
  Button,
  Dimensions,
  TouchableOpacity,
  SafeAreaView,
  FlatList,
  Image,
} from "react-native";
import { Auth } from "aws-amplify";
import { DataStore } from "@aws-amplify/datastore";
import { MementoModel } from "../../models";
import AppHeader from "../components/AppHeader";

const Item = ({ item, onPress }) => (
  <View>
    <TouchableOpacity onPress={onPress} style={styles.mementoListContainer}>
      <Image source={item.ProfileImage} resizeMode="cover" style={styles.image} />
      <Text style={styles.mementoTitle}>{item.Title}'s memento</Text>
      <Text style={styles.mementoDescription}>{item.Description}</Text>
    </TouchableOpacity>
  </View>
);

const { width, height } = Dimensions.get("window");

export default function Home({ navigation, updateAuthState }) {
  
  const [ mementoList, setMementoList ] = useState([]); 
  
  useEffect(() => {
    const ac = new AbortController();
    getMementos();
    return () => {ac.abort();}
  }, []);

  const getMementos = async () => {
    let mementos = await DataStore.query(MementoModel);
    setMementoList(mementos)
  };


  const _onPressItem = (item) => {
    navigation.navigate("MementoDetail", {item});
  };

  const renderItem = ({ item }) => {
    return <Item item={item} onPress={() => _onPressItem(item)} />;
  };

  async function signOut() {
    try {
      await Auth.signOut();
      updateAuthState("loggedOut");
    } catch (error) {
      console.log("Error signing out: ", error);
    }
  }

  return (
    <>
      <AppHeader />
      <View style={styles.container}>
        <View style={styles.mementoListContainer}>
          <View style={styles.mementoHeaderContainer}>
            <Text style={styles.mementoHeader}>my mementos</Text>
          </View>
          <FlatList
            data={mementoList}
            renderItem={renderItem}
            keyExtractor={(item) => item.id}
          />
        </View>
      </View>
      <View style={styles.buttonContainer}>
        <Button
          style={styles.signOut}
          title="Create memento"
          color="#50055E"
          onPress={() => navigation.navigate("CreateMemento")}
        />
        <Button
          style={styles.signOut}
          title="Sign Out"
          color="#50055E"
          onPress={signOut}
        />
      </View>
    </>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    marginTop: 20,
  },
  buttonContainer: {
    position: "absolute",
    flexDirection: "row",
    top: 80,
    width: width,
    height: 60,
    alignItems: "center",
    justifyContent: "space-between",
  },
  createMementoContainer: {
    position: "absolute",
    width: 150,
    height: 60,
    backgroundColor: "#50055E",
    borderRadius: 10,
    alignItems: "center",
    justifyContent: "center",
  },
  createMemento: {
    fontFamily: "serif",
    fontWeight: "600",
  },
  signOut: {
    position: "absolute",
    width: 150,
    height: 60,
    backgroundColor: "#50055E",
    borderRadius: 10,
    alignItems: "center",
    justifyContent: "center",
  },
  mementoHeaderContainer: {
    marginTop: 20,
    width: 300,
    height: 60,
    alignItems: "center",
    justifyContent: "center",
  },
  mementoHeader: {
    color: "#50055E",
    fontSize: 40,
  },
  mementoListContainer: {
    position: "relative",
    width: width,
    marginBottom: 20,
    alignItems: "center",
    justifyContent: "space-between",
  },
  mementoTitle: {
    top: 20,
    justifyContent: "flex-start",
    fontSize: 20,
  },
  mementoDescription: {
    position: "relative",
    fontSize: 15,
  },
  image: {
    width: 400,
    height: 300,
    marginTop: 10,
    borderRadius: 10,
  },
});
