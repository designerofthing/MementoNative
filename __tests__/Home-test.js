
import React from 'react';
import Home from '../src/screens/Home';
import renderer from 'react-test-renderer';

jest.setTimeout(15000);

it('renders correctly', async () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot('Home-test.js.snap');
});