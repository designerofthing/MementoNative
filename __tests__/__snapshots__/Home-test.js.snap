// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 90,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#50055E",
          "fontSize": 60,
        }
      }
    >
      memento
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "space-between",
          "marginBottom": 20,
          "position": "relative",
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 60,
            "justifyContent": "center",
            "marginTop": 20,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#50055E",
              "fontSize": 40,
            }
          }
        >
          my mementos
        </Text>
      </View>
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "position": "absolute",
        "top": 80,
        "width": 750,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#50055E",
              },
            ]
          }
        >
          Create memento
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#50055E",
              },
            ]
          }
        >
          Sign Out
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`renders correctly: Home-test.js.snap 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 90,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#50055E",
          "fontSize": 60,
        }
      }
    >
      memento
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "space-between",
          "marginBottom": 20,
          "position": "relative",
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 60,
            "justifyContent": "center",
            "marginTop": 20,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#50055E",
              "fontSize": 40,
            }
          }
        >
          my mementos
        </Text>
      </View>
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "position": "absolute",
        "top": 80,
        "width": 750,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#50055E",
              },
            ]
          }
        >
          Create memento
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#50055E",
              },
            ]
          }
        >
          Sign Out
        </Text>
      </View>
    </View>
  </View>,
]
`;
